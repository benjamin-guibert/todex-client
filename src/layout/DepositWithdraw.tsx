import React, { FC, useContext } from 'react'
import { ExchangeContext } from './ExchangeContext'
import StyledDepositWithdraw from './DepositWithdraw.style'
import DepositWithdrawInput from 'components/DepositWithdrawInput'
import { BigNumber } from 'ethers'

const DepositWithdraw: FC = () => {
  const { depositEther, depositToken, getTokenAllowance, approveToken } = useContext(ExchangeContext)

  const isTokenDepositAllowed = async (amount: BigNumber) => {
    const allowance = await getTokenAllowance()

    return !!allowance?.gte(amount)
  }

  return (
    <StyledDepositWithdraw>
      <DepositWithdrawInput name="ETH" onDeposit={depositEther} />
      <DepositWithdrawInput
        name="TDX"
        isDepositAllowed={isTokenDepositAllowed}
        onDeposit={depositToken}
        onApprove={approveToken}
      />
    </StyledDepositWithdraw>
  )
}

export default DepositWithdraw
