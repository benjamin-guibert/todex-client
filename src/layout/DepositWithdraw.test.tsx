import * as React from 'react'
import DepositWithdraw from './DepositWithdraw'
import renderer from 'react-test-renderer'

const renderComponent = () => renderer.create(<DepositWithdraw />)

describe('<DepositWithdraw />', () => {
  it('should render', () => {
    const result = renderComponent()

    expect(result.toJSON()).toMatchSnapshot()
  })
})
