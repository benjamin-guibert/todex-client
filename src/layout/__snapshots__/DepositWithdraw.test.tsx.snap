// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DepositWithdraw /> should render 1`] = `
<div
  className="sc-bdvvtL BjOxt container"
>
  <div
    className="m-3 px-2 py-1 card"
  >
    <form
      className=""
    >
      <div
        className="my-3 row"
      >
        <div
          className="input-group has-validation"
        >
          <span
            className="input-group-text"
          >
            ETH
          </span>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="my-3 row"
      >
        <div
          className="btn-group"
          role="group"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Deposit
          </button>
          <button
            className="btn btn-primary"
            disabled={true}
            type="button"
          >
            Withdraw
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    className="m-3 px-2 py-1 card"
  >
    <form
      className=""
    >
      <div
        className="my-3 row"
      >
        <div
          className="input-group has-validation"
        >
          <span
            className="input-group-text"
          >
            TDX
          </span>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="my-3 row"
      >
        <div
          className="btn-group"
          role="group"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Deposit (Approve)
          </button>
          <button
            className="btn btn-primary"
            disabled={true}
            type="button"
          >
            Withdraw
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
